### PR TITLE
add detail error when failed to create resource

### PR DESCRIPTION
**Pull request by wangchenglong01**
#### What type of this PR
- Code Style

/flake

### Which problem is this PR solving?
- add detail error when failed to create resource

### CODE REVIEW

File: pkg/controller/init.go [hunk 1]
```diff
@@ -185,6 +185,10 @@ func (c *Controller) initNodeSwitch() error {
 	}
 
 	_, err = c.config.KubeOvnClient.KubeovnV1().Subnets().Create(context.Background(), &nodeSubnet, metav1.CreateOptions{})
+	if err != nil {
+		klog.Errorf("failed to create subnet %s: %v", c.config.NodeSwitch, err)
+		return err
+	}
```
Comment (zhangzujian) [inline pkg/controller/init.go:191]: No need to return here. Or `return nil` in the following line.

File: pkg/controller/init.go [hunk 2]
```diff
@@ -491,6 +499,10 @@ func (c *Controller) initDefaultVlan() error {
 	}
 
 	_, err = c.config.KubeOvnClient.KubeovnV1().Vlans().Create(context.Background(), &defaultVlan, metav1.CreateOptions{})
+	if err != nil {
+		klog.Errorf("failed to create vlan %s: %v", defaultVlan, err)
+		return err
+	}
```
Comment (zhangzujian) [inline pkg/controller/init.go:505]: Same.

File: pkg/controller/init.go [hunk 3]
```diff
@@ -456,6 +460,10 @@ func (c *Controller) initDefaultProviderNetwork() error {
 	}()
 
 	_, err = c.config.KubeOvnClient.KubeovnV1().ProviderNetworks().Create(context.Background(), &pn, metav1.CreateOptions{})
+	if err != nil {
+		klog.Errorf("failed to create provider network %s: %v", c.config.DefaultProviderName, err)
+		return err
+	}
```
Comment (zhangzujian) [inline pkg/controller/init.go:466]: Same.

File: pkg/controller/init.go (hunk 1 above)
Comment (wangchenglong01) [reply pkg/controller/init.go:191]: @zhangzujian You're right. I have changed them and the latest code has been commited.

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**